Framework/API/src/Algorithm.cpp

```diff
@@ -2050,6 +2050,19 @@ Property *Algorithm::getPointerToPropertyOrdinal(const int &index) const {
  */
 const std::vector<Property *> &Algorithm::getProperties() const { return m_properties.getProperties(); }
 
+/**
+ * Return the list of declared property names.
+ * @return A vector holding strings of property names
+ */
+std::vector<std::string> Algorithm::getDeclaredPropertyNames() const noexcept {
+  std::vector<std::string> names;
```

Comment:
I took a quick look as it was in the approved list for a bit. Could I have some help in understanding if this can just call through to the  implementation in the property manager, like the other methods that are implementing that interface?

Reply:
@DavidFair is probably the best person to answer this

Reply:
That's correct, the inheritance is a bit unusual but I didn't want to start refactoring core components in a fairly big PR.

Basically as a diagram the inheritance chain looks something like:
```
                       -  - - [ PropertyManager ] 
| IPropertyManager| - |  -  - [ IAlgorithm/Algorithm ]
                      -  - -  [ PropertyManagerOwner ]
```

This means that all the (non-templated) methods are duplicated between Algorithm / PropertyManager, instead of being inherited. This can be seen in other methods such as [Algorithm::getProperties](https://github.com/mantidproject/mantid/blob/main/Framework/API/src/Algorithm.cpp#L2051) and [PropertyManager::getProperties](https://github.com/mantidproject/mantid/blob/main/Framework/Kernel/src/PropertyManager.cpp#L660)

It would be nice to fix the inheritance chain to
```
[IPropertyManager]  - - - [ PropertyManager] - [IAlgorithm / Algorithm]
                    - - - [ PropertyManagerOwner ]
```
and completely remove all the code duplication which does identical ops in 90% of cases, but this would be a core / extra attention PR if we did it

Reply:
Yes the owner thing is a little odd and I'm happy not doing a bigger change here now. My main question here really was whether this method can just do `m_properties.getDeclaredPropertyNames()` like `getProperties` does to remove duplication of the implementation?

Reply:
Ah I hadn't spotted that it has a PropertyManager (I assumed is-a), we could definitely do that 👍